src/linear_algebra/finsupp.lean

```diff
@@ -359,6 +360,28 @@ lemma lift_apply (f) (g) :
   ((lift M R X) f) g = g.sum (λ x r, r • f x) :=
 rfl
 
+/-- Given an `S`-algebra `R`, an `R`-module `M` with a compatible `S`-module structure, and a
+type `X`, then the set of functions `X → M` is `S`-linearly equivalent to the `R`-linear maps from
+the free `R`-module on `X` to `M`. -/
+noncomputable def llift (S : Type*) [comm_semiring S] [algebra S R]
```

Comment:
Do you actually need `algebra S R` here?

Reply:
Ooh, my bad, I don't! In fact when I use this elsewhere in the PR I just have R = S...